### ISSUE TITLE
Some Mods don't work and Vanilla Game features are broken as theseus ships JRE instead of JDK

### ISSUE THREAD

**Issue by Dev0Louis** [modrinth/code#1192]
### Describe the bug

As the Modrinth launcher uses a JRE by default instrumentation isn't available, but some mods do rely on it to work. 
Examples are: https://modrinth.com/mod/tameable-axolotls and https://modrinth.com/mod/impirium.

### Steps to reproduce

1. Create new profile with the default java from theseus
2. Add Tamable Axolotls or Impirium
3. Launch the Game
4. Watch it crash

- Fix -
5. Change Java to JDK
6. Launch Game
7. Now it works 🎉 

### Expected behavior

Theseus should ship a JDK by default as that is what most users have and is required for some extreme modifications of Minecraft. 

### System information

Linux, but that should be irrelevant

**Comment by brawaru** [comment]
Because Minecraft runs with just JRE, this should be the target for Minecraft mods. If you want to use mods that rely on APIs meant for developers, you are welcome to install a JDK yourself. Most users don't need that.

**Comment by Gaming32** [comment]
This is true for 1.16.x and older. Since 1.17, the game uses a full JDK. While it doesn't use any JDK-specific features *yet*, they officially run on a full JDK. If the Modrinth launcher doesn't do that (like the official launcher does), it's just setting itself up for incompatibilities like these.

**Comment by Gaming32** [comment]
Additional comment: the Vanilla game does actually use the JDK. The absence of it might make users confused why the `/jfr` command throws an error on the Modrinth launcher (and only on the Modrinth launcher).

**Comment by blryface** [comment]
I don't think the mods linked in the issue are meant to only run on a dev env, and I don't see how or why a mod would be specifically made for a dev env? It may have features that help you develop but it wouldn't make that much sense to make them "only for" a dev env

**Comment by Felix14-v2** [comment]
Here's an example from my bookmarks: https://github.com/EZForever/ThatOrThis/pull/27#issuecomment-1609111562
Also iirc CaffeineMC used a mod which allows them to freeze some parts of the rendering engine for debug purposes, but I can't find it

**Comment by Dev0Louis** [comment]
> Some mods are made only for development environment, but we shouldn't turn the launcher into a DE because of these specific mods. However, breaking the vanilla functionality is unacceptable for the launcher, so I support the suggestion.

None of the mods are for "dev env" its just that there are some features in the JDK (Instruments) for runtime class redefinition which are not present in the JRE. These features need to be present for some mods to work.

**Comment by brawaru** [comment]
I wasn't aware that the latest MoJRE is a JDK, so yeah, for compatibility reasons it would be better to map MoJRE 21 to Azul's JDK package, but that's because it seems that Modrinth App is breaking some of the vanilla functionality like `/jfr` command[^1], not because of the mods in question that override superclasses in a seemingly cursed way[^2] with dependence on development APIs.

[^1]: https://discord.com/channels/734077874708938864/1061855024252207167/1240516520057110620
[^2]: https://github.com/ekulxam/TameableAxolotls/blob/84d1813ebe163e0cf43946f0cc1f506211bcea85/src/main/java/survivalblock/tameable_axolotls/TameableAxolotlsAgent.java#L21

**Comment by Dev0Louis** [comment]
> I wasn't aware that the latest MoJRE is a JDK, so yeah, for compatibility reasons it would be better to map MoJRE 21 to Azul's JDK package, but that's because it seems that Modrinth App is breaking some of the vanilla functionality like `/jfr` command[1](#user-content-fn-1-2a1cd9a7d048778b424d20d349c8955c), not because of the mods in question that override superclasses in a seemingly cursed way[2](#user-content-fn-2-2a1cd9a7d048778b424d20d349c8955c) with dependence on development APIs.
> ## Footnotes
> 
>     1. https://discord.com/channels/734077874708938864/1061855024252207167/1240516520057110620 [↩](#user-content-fnref-1-2a1cd9a7d048778b424d20d349c8955c)
> 
>     2. https://github.com/ekulxam/TameableAxolotls/blob/84d1813ebe163e0cf43946f0cc1f506211bcea85/src/main/java/survivalblock/tameable_axolotls/TameableAxolotlsAgent.java#L21 [↩](#user-content-fnref-2-2a1cd9a7d048778b424d20d349c8955c)

Well but there is not other way that I know of to change the inheritance structure of Classes. And do what has been done Another example of a mod using this is impirium(https://github.com/Bawnorton/Impirium/blob/main/src/main/java/com/bawnorton/impirium/ImPIriumAgent.java) as I pointed out above.

That behavior is supported by all other launchers, including the vanilla Default launcher, so it should be too on Modrinth launcher without any tinkering. 

And why only 21? 17 is just as affected by this.

**Comment by Sl3epy** [comment]
Has there been any update to this?
